src/OrchardCore.Modules/OrchardCore.Lists/Services/ContainerService.cs

```diff
@@ -339,9 +355,43 @@ public async Task<IEnumerable<ContentItem>> QueryContainedItemsAsync(string cont
             }
         }
 
-        private static Expression<Func<ContentItemIndex, bool>> CreateOrderedContentIndexFilter(bool publishedOnly)
+        private IQuery<ContentItem> ContainedItemOptionsFilter(ContainedItemOptions containedItemOptions, IQuery<ContentItem> containedItems)
+        {
+            if (!string.IsNullOrEmpty(containedItemOptions.DisplayText))
+            {
+                containedItems.With<ContentItemIndex>().Where(i => i.DisplayText.Contains(containedItemOptions.DisplayText));
+            }
+
+            if (containedItemOptions.Status != ContentsStatus.None)
+            {
+                switch (containedItemOptions.Status)
+                {
+                    case ContentsStatus.Draft:
+                        containedItems.With<ContentItemIndex>(i => !i.Published);
```

Comment:
This doesnt work, it has to be `i => !i.Published && i.Latest` or you get all history

(I'm sure I already said this)